ouroboros-consensus-cardano/tools/db-analyser/Main.hs

```diff
@@ -73,15 +86,30 @@ parseCmdLine = CmdLine
             long "verbose"
           , help "Enable verbose logging"
           ])
-    <*> switch (mconcat [
-            long "onlyImmutableDB"
-          , help "Validate only the Immutable DB (e.g. do not do ledger validation)"
-          ])
+    <*> parseSelectDB
     <*> parseValidationPolicy
     <*> blockTypeParser
     <*> parseAnalysis
     <*> parseLimit
 
+parseSelectDB :: Parser SelectDB
```

Comment:
What does the `--help` look like with this? Is there any indication that they cannot pass `analyse-from` unless they pass `only-immutable-db`?

I anticipate it's necessary to encode that requirement explicitly, as an ad-hoc pass that happens immediately after we parse the command line.

Reply:
I've spent some time on this so that it looks exactly as you would hope it to look :)

If you run the tool with no arguments you get

```
cabal run ouroboros-consensus-cardano:db-analyser
Temporary modify `cabal.project` to use nix builds of `source-repository-package`s (haskell/cabal#5444 and haskell/cabal#6249).
Up to date
Missing: --db PATH COMMAND

Usage: db-analyser --db PATH [--verbose]
                   [--only-immutable-db [--analyse-from SLOT_NUMBER]]
                   [--validate-all-blocks | --minimum-block-validation] COMMAND
                   [--show-slot-block-no | --count-tx-outputs |
                     --show-block-header-size | --show-block-txs-size |
                     --show-ebbs | --store-ledger SLOT_NUMBER]
                   [--num-blocks-to-process INT]
```

If you provide the `--analyse-from` but forget to mention to run on immutable db only you get a nice error


```
cabal run ouroboros-consensus-cardano:db-analyser -- --analyse-from 1
Temporary modify `cabal.project` to use nix builds of `source-repository-package`s (haskell/cabal#5444 and haskell/cabal#6249).
Up to date
Missing: --db PATH --only-immutable-db COMMAND

Usage: db-analyser --db PATH [--verbose]
                   [--only-immutable-db [--analyse-from SLOT_NUMBER]]
                   [--validate-all-blocks | --minimum-block-validation] COMMAND
                   [--show-slot-block-no | --count-tx-outputs |
                     --show-block-header-size | --show-block-txs-size |
                     --show-ebbs | --store-ledger SLOT_NUMBER]
                   [--num-blocks-to-process INT]
```